### PR TITLE
Added the update functionality and tests to cover it.

### PR DESCRIPTION
**Pull request by bumblebot1**

### CODE REVIEW

File: public/main.js [hunk 1]
```diff
@@ -80,4 +91,22 @@ function deleteEntry(todo) {
     createRequest.send();
 }
 
+function completeEntry(todo) {
+    var createRequest = new XMLHttpRequest();
+    createRequest.open("PUT", "/api/todo/");
+    createRequest.setRequestHeader("Content-type", "application/json");
+    createRequest.onload = function() {
```
Comment (dwybourn) [inline public/main.js]: This onload() method is exactly the same as the one in deleteEntry(), but with a different error message. Best to refactor this into a method that takes in an error message as the argument.

Comment (bumblebot1) [inline public/main.js]: On it

File: public/main.js [hunk 2]
```diff
@@ -60,7 +63,15 @@ function reloadTodoList() {
                 deleteEntry(todo);
             });
 
+            var completeButton = document.createElement("button");
```
Comment (kenneth-gray) [inline public/main.js]: I think this has got to the point where we have 2 buttons, so refactoring out to a function that creates buttons would be really good.

File: public/main.js [hunk 3]
```diff
@@ -52,6 +53,8 @@ function reloadTodoList() {
         todos.forEach(function(todo) {
             var listItem = document.createElement("li");
             listItem.textContent = todo.title;
+            listItem.style.fontStyle = todo.complete === true ? "italic" : "normal";
+            listItem.style.fontWeight = todo.complete === true ? "normal" : "bold";
```
Comment (lclarksl) [inline public/main.js]: It would be better if you added a 'complete' class and then did the styling for this in the css.

File: public/main.js [hunk 4]
```diff
@@ -52,15 +53,14 @@ function reloadTodoList() {
         todos.forEach(function(todo) {
             var listItem = document.createElement("li");
             listItem.textContent = todo.title;
+            listItem.className = todo.complete === true ? "complete" : "incomplete";
```
Comment (kenneth-gray) [inline public/main.js]: If `todo.complete` is a boolean, no need to check against `true`

File: public/main.js [hunk 5]
```diff
@@ -71,13 +71,40 @@ function deleteEntry(todo) {
...
+    var button = document.createElement("button");
+    button.textContent = name;
+    button.className = style;
+    button.addEventListener("click", function() {
+        listener(todo);
+    });
+    return button;
+}
+
+function loader(err, status) {
```
Comment (kenneth-gray) [inline public/main.js]: Suggestion:

```
function onloadFactory(err, httpStatusCode) {
    return function() {
        // Rest of code inside original function
    };
}
```

It just removes extra lines

File: public/main.css [hunk 6]
```diff
@@ -30,3 +30,20 @@ h1 {
     background-color: rgba(52, 114, 162, 1);
     font-family: cursive;
 }
+
+.completeButton{
```
Comment (kenneth-gray) [inline public/main.css]: `complete-button` and `delete-button`

**Comment by kenneth-gray** [comment]
:shipit:

**Pull request merged**